datalad/interface/tests/test_run_procedure.py

```diff
@@ -304,17 +305,52 @@ def test_quoting(path):
         with assert_raises(CommandError):
             runner.run("datalad run-procedure just2args 'still-one arg'")
 
+
 @skip_if_on_windows
-@with_tempfile
-def test_text2git_empty(path):
-    """
-    Tests that empty files are not annexed in a ds configured with text2git.
-    """
+@with_tree(tree={
+    # "TEXT" ones
+    'empty': '',  # we have special rule to treat empty ones as text
+    # check various structured files - libmagic might change its decisions which
+    # can effect git-annex. https://github.com/datalad/datalad/issues/3361
+    'JSON': """\
+{
+    "name": "John Smith",
+    "age": 33
+}
+""",
+    'YAML': """\
+--- # The Smiths
+- {name: John Smith, age: 33}
+- name: Mary Smith
+  age: 27
+""",
+    'MARKDOWN': """\
+# Title
+
+## Section1
+
+When the earth was flat
+
+## Section2
+""",
+    # BINARY ones
+    '0blob': '\x00',
+    'emptyline': '\n',  # libmagic: "binary" "application/octet-stream"
+})
+def test_text2git(path):
+    # Test if files being correctly annexed in a ds configured with text2git.
+    TEXT_FILES = ('JSON', 'YAML', 'MARKDOWN', 'empty')
+    BINARY_FILES = ('0blob', 'emptyline')
+
     ds = Dataset(path).create(force=True)
     ds.run_procedure('cfg_text2git')
+
```

Comment:
This space is another change that isn't from either side.

Reply:
see comment below

Reply:
I don't see how the comment below is relevant.  Even if the test was rewritten on 0.11.x's side (specifically da27e86), the resolution shouldn't introduce a space that wasn't in 0.11.x or in master.

Reply:
oh, gotcha. True!  Let's see if we agree on the below and then I could rewrite that merge to make it clearer (to avoid possible future conflicts)... or you are most welcome to do so and force push here!

Reply:
> you are most welcome to do so and force push here!

OK, I'll do that.